docs/platform_developers/developer_getting_started.rst

```diff
@@ -0,0 +1,118 @@
+********************************
+Developer Getting Started Guide
+********************************
+
+This page describes how to setup your development environment to:
+
+* Install VS Code and Inmanta extension.
+* Setting up Python virtual environments.
+* Benefit from linting and code navigation by setting up a project.
+* Set project sources
+* Module developers guide
+
+
+Install VS Code and Inmanta extension
+#######################################
+
+This step does not need any additional information but if you need some help, you can always ask a colleague.
+
+In order to install VS Code, you can refer to `this <https://code.visualstudio.com/learn/get-started/basics>`_ page.
+
+Further information about Inmanta VS Code extension is available on `this <https://github.com/inmanta/vscode-inmanta>`_ page.
+
+
+Setting up Python virtual environments
+########################################
+
+Python virtual environments have to be created using ``venv`` by running:
+
+.. code-block:: bash
+    
+    python3 -m venv ~/.virtualenvs/<env_name>
+
+Then activated by running:
+
+.. code-block:: bash
+    
+    source ~/.virtualenvs/<env_name>/bin/activate
+
+**Upgrading your ``pip`` will save you a lot of time and troubleshooting (due to changes in the pip resolver in version 20 and 21).** you can do so by running:
+
+.. code-block:: bash
+    
+    pip install --upgrade pip
+
+Usually the projects that you will work on, come with ``requirements.txt`` or ``requirements.dev.txt`` to install the required modules, if not, it is a good idea to ``pip install``:
+
+1. inmanta-core
+2. pytest
+3. pytest-inmanta
+4. pytest-inmanta-* (Additional pytest fixtures exist for working with yang and lsm, but are only useful when these are used in your project or module.)
+
+
+Benefit from linting and code navigation by setting up a project
+##################################################################
+
+At the time of this writing, linting and code navigation in IDEs work only if you have a project, so if you want to work on a single module, you still need to have a project.
+
+There are two things that can be done:
+
+1. Either there is already a project and you ``git clone`` your module there.
+2. Make a project.
+
+Steps to create a project are mentioned `here <https://docs.inmanta.com/community/latest/model_developers/configurationmodel.html>`_ for further reading.
```

Comment:
I would put the cookie cutter instructions here as well.

Reply:
Done